src/randgraphs.jl

```diff
@@ -193,6 +193,191 @@ function barabasi_albert(n::Integer, k::Integer; is_directed=false, seed::Int=-1
     return g
 end
 
+"""
+    static_fitness_model{T<:Real}(m::Int, fitness::Vector{T}; seed::Int=-1)
+
+Generates a non-growing random undirected graph with `length(fitness)` nodes and `m` edges, 
+in which the edge probabilities proportional to node fitness scores. 
+Time complexity is O(|V| + |E| log |E|).
+
+Reference:
+
+* Goh K-I, Kahng B, Kim D: Universal behaviour of load distribution 
+in scale-free networks. Phys Rev Lett 87(27):278701, 2001.
+"""
+function static_fitness_model{T<:Real}(m::Int, fitness::Vector{T}; seed::Int=-1)
+    @assert(m >= 0, "invalid number of edges")
+    n = length(fitness)
+    if m == 0
+        return Graph(n)
+    end
+    # sanity check for the fitness
+    @assert(minimum(fitness) >= zero(eltype(fitness)), "fitness scores must be non-negative")
+    # avoid getting into an infinite loop when too many edges are requested
+    nodes = 0
+    for f in fitness
+        if f > zero(f)
+            nodes += 1
+        end
+    end
+    max_no_of_edges = div(nodes*(nodes-1),2)
+    @assert(m <= max_no_of_edges, "too many edges requested")
+    # calculate the cumulative fitness scores
+    cum_fitness = cumsum(fitness)
+    max_fitness = cum_fitness[end]
+    edges = Set{Edge}()
+    rng = getRNG(seed)
+    while m > 0
+        source = searchsortedfirst(cum_fitness, rand(rng)*max_fitness)
+        target = searchsortedfirst(cum_fitness, rand(rng)*max_fitness)
+        # skip if loop edge
+        (source == target) && continue
+        if source > target
+            source, target = target, source
+        end
+        edge = Edge(source, target)
+        # is there already an edge? If so, try again
+        in(edge, edges) && continue
+        # insert the edge
+        push!(edges, edge)
+        m -= 1
+    end
+    # create the graph
+    g = Graph(n)
+    for edge in edges
+        add_edge!(g, edge)
+    end
+    return g
+end
+
+function static_fitness_model{T<:Real,S<:Real}(m::Int, fitness_out::Vector{T}, fitness_in::Vector{S}; seed::Int=-1)
+    @assert(m >= 0, "invalid number of edges")
+    n = length(fitness_out)
+    @assert(length(fitness_in)==n, "fitness_in must have the same size as fitness_out")
+    if m == 0
+        return DiGraph(n)
+    end
+    # sanity check for the fitness
+    @assert(minimum(fitness_out) >= zero(eltype(fitness_out)) && minimum(fitness_in) >= zero(eltype(fitness_in)), "fitness scores must be non-negative")
+    # avoid getting into an infinite loop when too many edges are requested
+    outnodes = 0
+    innodes = 0
+    nodes = 0
+    for i in eachindex(fitness_out)
```

Comment:
Why `eachindex`? For 1-dimensional arrays, `1:length(fitness_out)` is at least equivalent, if not more efficient.